src/NServiceBus.Core/Pipeline/Outgoing/RoutingToDispatchConnector.cs

```diff
@@ -52,7 +40,34 @@ public override Task Invoke(IRoutingContext context, Func<IDispatchContext, Task
             return stage(this.CreateDispatchContext(operations, context));
         }
 
+        static void LogOutoingOperationsWhenDebugEnabled(TransportOperation[] operations)
+        {
+            if (!isDebugEnabled)
```

Comment:
If you take my advice around changing `isDebugEnabled`, you'd want to remove this `if` statement and rename the method.